### PR TITLE
ui: Thread through data-source invalidate method

### PR DESCRIPTION
**Pull request by johncowen**
### Description
This PR threads through an `invalidate` method from `DataSource` through to `DataLoader` and exposes it there.

The `invalidate` method always existed but we've never used it/needed it, and is used to notify the DataSource/Loader that the data associated with it is invalidated. The result of doing this means that the DataSource will immediately try to request a new set of data.

A little background here: As we usually use our blocking query support, we've never really needed to invalidate the data associated with DataSources as blocking queries always keep our data fresh.

As we are using more and more traditional polling for UI liveness, data could be out of date during the interval of the poll, by using `invalidate` you can force an immediate request for data, which will continue to poll at the same interval.

TLDR; `invalidate` _almost_ takes the DataSource component off the page and puts it back again, thus instantiating it all over again.

I left in the console log reminder, but turned it to debug for when we want to implement the additional feature mentioned in the debug log.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

/cc @LevelbossMike

### CODE REVIEW

**Review by LevelbossMike — COMMENTED** [review]
Added a comment about the statechart that is driving this component. We seem to work around that now.

When would we need to invalidate the data-loader from a UI perspective?

File: ui/packages/consul-ui/app/components/data-loader/index.js [hunk 1]
```diff
@@ -22,12 +21,6 @@ export default Component.extend(Slotted, {
     this.dispatch('LOAD');
   },
   actions: {
-    invalidate() {
-      this.dispatch('INVALIDATE');
```
[the remark this answers is not in the document]
Comment (johncowen) [reply ui/packages/consul-ui/app/components/data-loader/index.js:26]: Oh that state should have been removed here also! Thanks for spotting, I've removed it also aswell as the action.

**Review by evrowe — APPROVED** [review]
